filters/builtin/stripquery.go

```diff
@@ -51,7 +51,8 @@ func validHeaderFieldByte(b byte) bool {
 }
 
 // make sure we don't generate invalid headers
-func sanitize(input string) string {
+// temporary public function to benchmark it
```

Comment:
There is no need to export it as benchmarks sits in the same package.
Usually we introduce benchmark as a first commit and include baseline benchmark results in the commit message.
Then change implementation, re-run benchmark and compare results to the baseline via https://pkg.go.dev/golang.org/x/perf/cmd/benchstat

To get a statistically meaningful benchmark results it should be run several times, usually we run 10.
See this PR as an example of performance-related change https://github.com/zalando/skipper/pull/2870

I've also created a small script to automate benchmarking two commits (it important that benchmark function is added in a separate commit to establish the baseline), see https://github.com/golang/go/issues/63233#issuecomment-1736468193